### PR TITLE
Clarify content-type requirements

### PR DESCRIPTION
**Pull request by dichotommy**

### CODE REVIEW

File: reference/api/README.md [hunk 1]
```diff
@@ -14,15 +14,13 @@ Check out [the FAQ](/resources/faq.md) for answers to some common questions 💡
 
 #### Content type
 
- Requests can be in JSON, CSV, or NDJSON. Responses are always in JSON.
-
-MeiliSearch currently supports the following formats:
+Any endpoint with a payload (`--data-binary`) requires a `Content-Type` header. MeiliSearch currently supports the following formats:
```
Comment (dichotommy) [inline reference/api/README.md:17]:
```suggestion
Any API request with a payload (`--data-binary`) requires a `Content-Type` header. MeiliSearch currently supports the following formats:
```

File: reference/api/README.md [hunk 2]
```diff
@@ -14,15 +14,13 @@ Check out [the FAQ](/resources/faq.md) for answers to some common questions 💡
...
-
-MeiliSearch currently supports the following formats:
+Any API request with a payload (`--data-binary`) requires a `Content-Type` header. MeiliSearch currently supports the following formats:
 
 - `Content-Type: application/json` for JSON
 - `Content-Type: application/x-ndjson` for NDJSON
 - `Content-Type: text/csv` for CSV
 
-You **don't need** to specify a header for `GET` and `DELETE` routes. Routes that require a payload **only accept JSON headers.** `PUT` and `POST` document routes accept all headers.
+However, the [add documents](/reference/api/documents.md#add-or-replace-documents) and [update documents](/reference/api/documents.md#add-or-update-documents) endpoints are the only ones which currently accept NDJSON and CSV `Content-Type`. For all other endpoints, use `Content-Type: application/json`.
```
Comment (dichotommy) [inline reference/api/README.md:23]:
```suggestion
However, the only endpoints that currently accept NDJSON and CSV `Content-Type` are the [add documents](/reference/api/documents.md#add-or-replace-documents) and [update documents](/reference/api/documents.md#add-or-update-documents) endpoints. For all others, use `Content-Type: application/json`.
```

**Review by maryamsulemani97 — APPROVED** [review]
⛄️

File: reference/api/README.md [hunk 3]
```diff
@@ -14,15 +14,13 @@ Check out [the FAQ](/resources/faq.md) for answers to some common questions 💡
...
-
-MeiliSearch currently supports the following formats:
+Any API request with a payload (`--data-binary`) requires a `Content-Type` header. MeiliSearch currently supports the following formats:
 
 - `Content-Type: application/json` for JSON
 - `Content-Type: application/x-ndjson` for NDJSON
 - `Content-Type: text/csv` for CSV
 
-You **don't need** to specify a header for `GET` and `DELETE` routes. Routes that require a payload **only accept JSON headers.** `PUT` and `POST` document routes accept all headers.
+However, the only endpoints which currently accept NDJSON and CSV `Content-Type` are the [add documents](/reference/api/documents.md#add-or-replace-documents) and [update documents](/reference/api/documents.md#add-or-update-documents) endpoints. For all others, use `Content-Type: application/json`.
```
Comment (guimachiavelli) [inline reference/api/README.md:23]: Nitpick: I feel this "However" only makes the sentence more complex without providing any benefits.

```suggestion
The only endpoints which currently accept NDJSON and CSV `Content-Type` are the [add documents](/reference/api/documents.md#add-or-replace-documents) and [update documents](/reference/api/documents.md#add-or-update-documents) endpoints. For all others, use `Content-Type: application/json`.
```

**Review by maryamsulemani97 — APPROVED** [review]

**Comment by dichotommy** [comment]
bors merge

**Pull request merged**